lago/utils.py

```diff
@@ -533,12 +534,12 @@ def load_virt_stream(virt_fd):
         dict: Loaded virt config
     """
     try:
-        virt_conf = json.load(virt_fd)
+        virt_conf = json.load(virt_fd, object_pairs_hook=collections.OrderedDict)
     except ValueError:
         virt_fd.seek(0)
         virt_conf = yaml.load(virt_fd)
 
-    return deepcopy(virt_conf)
+    return copy.deepcopy(virt_conf)
```

Comment:
On 05/24 04:34, Yaniv Kaul wrote:
> >      except ValueError:
> >          virt_fd.seek(0)
> >          virt_conf = yaml.load(virt_fd)
> >  
> > -    return deepcopy(virt_conf)
> > +    return copy.deepcopy(virt_conf)
> 
> Indeed, but the original (copy.deepcopy() function) seems to do the work here.

That's because we are not using the yaml anchors[1] anywhere, if you did, it
would break (as the shared dicts and lists are shared as links to the same,
and not independent copies).
That was the only reason to run deepcopy there, as I said, probably the naming
should have been better, something like 'remove_internal_cross_objects' or
something more verbose would have been better.


[1] https://en.wikipedia.org/wiki/YAML#Repeated_nodes
> 
> ---
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/lago-project/lago/pull/262/files/9e9c7b1a9a17511353aeca189f7ebe75ee61ed4a#r64374333

-- 
David Caro

Red Hat S.L.
Continuous Integration Engineer - EMEA ENG Virtualization R&D

Tel.: +420 532 294 605
Email: dcaro@redhat.com
IRC: dcaro|dcaroest@{freenode|oftc|redhat}
Web: www.redhat.com
RHT Global #: 82-62605